### PR TITLE
Bugfix/iia 907 the done button for add fqn and add other name should be disabled until all the data is entered in the screen

### PR DESCRIPTION
**Pull request by swaroopsalvi**
Fixed Done button on Pattern screen.

### CODE REVIEW

File: kview/src/main/java/dev/ikm/komet/kview/mvvm/view/descriptionname/DescriptionNameController.java [hunk 1]
```diff
@@ -103,47 +107,52 @@ public DescriptionNameController() { }
...
 
         setupComboBox(nameDescriptionType, fetchDescriptionTypes()); // Hard coded
-        nameDescriptionType.valueProperty().bind(descrNameViewModel.getProperty(NAME_TYPE));
+        ObjectProperty<ConceptEntity> nameTypeProp = descrNameViewModel.getProperty(NAME_TYPE);
+        nameDescriptionType.valueProperty().bind(nameTypeProp);
+        nameTypeProp.addListener(fieldsValidationListner);
 
-        editDescriptionTitleLabel.textProperty().bind(descrNameViewModel.getProperty(TITLE_TEXT));
-        nameTextField.textProperty().bindBidirectional(descrNameViewModel.getProperty(NAME_TEXT));
+        SimpleStringProperty nameTextProp = descrNameViewModel.getProperty(NAME_TEXT);
```
Comment (carldea) [inline kview/src/main/java/dev/ikm/komet/kview/mvvm/view/descriptionname/DescriptionNameController.java:130]: Can the statement be more abstract? 
`StringProperty nameTextProp = ...;`

File: kview/src/main/java/dev/ikm/komet/kview/mvvm/view/descriptionname/DescriptionNameController.java [hunk 2]
```diff
@@ -103,47 +107,52 @@ public DescriptionNameController() { }
 
     @FXML
     public void initialize() {
+        ChangeListener fieldsValidationListner = (obs, oldValue, newValue) -> {
```
Comment (carldea) [inline kview/src/main/java/dev/ikm/komet/kview/mvvm/view/descriptionname/DescriptionNameController.java:110]: Misspelling on listener.

File: kview/src/main/java/dev/ikm/komet/kview/mvvm/view/pattern/PatternFieldsController.java [hunk 3]
```diff
@@ -114,6 +117,12 @@ public class PatternFieldsController {
 
     @FXML
     private void initialize() {
+        ChangeListener fieldsValidationListner = (obs, oldValue, newValue) -> {
```
Comment (carldea) [inline kview/src/main/java/dev/ikm/komet/kview/mvvm/view/pattern/PatternFieldsController.java:120]: Can you update the spelling to `fieldsValidationListener`?

File: kview/src/main/java/dev/ikm/komet/kview/mvvm/viewmodel/DescrNameViewModel.java [hunk 4]
```diff
@@ -62,20 +64,59 @@ public class DescrNameViewModel extends FormViewModel {
...
     public DescrNameViewModel() {
         super(); // defaults to View mode
-        addProperty(NAME_TEXT, "")
+                addProperty(NAME_TEXT, "")
+                .addValidator(NAME_TEXT, "Name Text", (ReadOnlyStringProperty prop, ViewModel vm) -> {
+                    if (prop.isEmpty().get()) {
+                        return new ValidationMessage(NAME_TEXT, MessageType.ERROR, "${%s} is required".formatted(NAME_TEXT));
+                    }
+                    return VALID;
+                })
```
Comment (carldea) [inline kview/src/main/java/dev/ikm/komet/kview/mvvm/viewmodel/DescrNameViewModel.java:77]: Can we use Cognitive's new feature to simplify verbose Validators and allows you to add more validation messages? (Note: the new feature was created in Cognitive 1.2.0)

Can you bump up there version of Cognitive 1.3.0?

The above code can be rewritten as the following:
```java
.addValidator(NAME_TEXT, "First Name", (ReadOnlyStringProperty prop, ValidationResult validationResult, ViewModel viewModel) -> {
    if (prop.isEmpty().get()) {
        validationResult.error("${%s} is required".formatted(NAME_TEXT));
    }
});
```

While you only have one validation it is still less verbose. The feature allows you to create multiple validation messages.

File: kview/src/main/resources/dev/ikm/komet/kview/mvvm/view/descriptionname/description-name.fxml [hunk 5]
```diff
@@ -63,22 +63,22 @@
                               <Insets right="15.0" />
                            </GridPane.margin>
                         </ComboBox>
-                            <ComboBox fx:id="caseSignificanceComboBox" maxWidth="1.7976931348623157E308" prefHeight="25.0" prefWidth="310.0" styleClass="blue-arrow-combo-box" promptText="Select case sensitivity" GridPane.columnIndex="1" GridPane.hgrow="ALWAYS" GridPane.rowIndex="2">
+                            <ComboBox fx:id="caseSignificanceComboBox" maxWidth="1.7976931348623157E308" prefHeight="25.0" prefWidth="310.0" promptText="Select case sensitivity" styleClass="blue-arrow-combo-box" GridPane.columnIndex="1" GridPane.hgrow="ALWAYS" GridPane.rowIndex="2">
```
Comment (carldea) [inline kview/src/main/resources/dev/ikm/komet/kview/mvvm/view/descriptionname/description-name.fxml:66]: Should the `prompText` be `Select case significance`?

**Review by carldea — COMMENTED** [review]
Looks very good. 
@swaroopsalvi If you don't mind I have a small request to improve the code a bit (it should not take long).

If you could make use of Cognitive's newer feature that allows validation code to be more concise (v1.3.0).

This helps improve code readability regarding validators and will update Cognitive's latest version.

File: kview/src/main/resources/dev/ikm/komet/kview/mvvm/view/descriptionname/description-name.fxml (hunk 5 above)
Comment (swaroopsalvi) [reply kview/src/main/resources/dev/ikm/komet/kview/mvvm/view/descriptionname/description-name.fxml:66]: Yes, I was looking at wrong FIGMA (concept one).

File: kview/src/main/java/dev/ikm/komet/kview/mvvm/viewmodel/DescrNameViewModel.java (hunk 4 above)
Comment (swaroopsalvi) [reply kview/src/main/java/dev/ikm/komet/kview/mvvm/viewmodel/DescrNameViewModel.java:77]: updated code to use 1.2.0 feature.
Updated pom.xml to use cognitive 1.3.0 version.

File: kview/src/main/java/dev/ikm/komet/kview/mvvm/view/descriptionname/DescriptionNameController.java (hunk 2 above)
Comment (swaroopsalvi) [reply kview/src/main/java/dev/ikm/komet/kview/mvvm/view/descriptionname/DescriptionNameController.java:110]: fixed.

File: kview/src/main/java/dev/ikm/komet/kview/mvvm/view/descriptionname/DescriptionNameController.java (hunk 1 above)
Comment (swaroopsalvi) [reply kview/src/main/java/dev/ikm/komet/kview/mvvm/view/descriptionname/DescriptionNameController.java:130]: fixed.

File: kview/src/main/java/dev/ikm/komet/kview/mvvm/view/descriptionname/DescriptionNameController.java [hunk 6]
```diff
@@ -154,28 +163,7 @@ private void handleCancelButtonEvent(ActionEvent actionEvent) {
...
-        boolean isLanguageComboBoxComboBoxSelected = languageComboBox.getValue() != null;
-
-        submitButton.setDisable(
-                isOtherNameTextFieldEmpty || !isModuleComboBoxSelected
-                || !isCaseSignificanceComboBoxSelected || !isLanguageComboBoxComboBoxSelected
-                || !isStatusComboBoxComboBoxSelected);
-    }
-
-    private void populateDialectComboBoxes() {
+     private void populateDialectComboBoxes() {
```
Comment (dholubek) [inline kview/src/main/java/dev/ikm/komet/kview/mvvm/view/descriptionname/DescriptionNameController.java:166]: can we remove this extra space in front of private?

**Review by carldea — APPROVED** [review]
Great job! 
Glad to see tinkar-core and cognitive libraries updated.

**Pull request merged**